#!/usr/bin/env python
PACKAGE = "two_d_guidance"
    
from dynamic_reconfigure.parameter_generator_catkin import *
    
gen = ParameterGenerator()
roi_grp = gen.add_group("RoI", type="tab")
roi_grp.add("roi_xc", int_t,    0, "Region of Interest center x(pixel)", 350,  0, 1024)
roi_grp.add("roi_yc", int_t,    0, "Region of Interest center y",        200,  0, 1024)
roi_grp.add("roi_dx", int_t,    0, "Region of Interest width",           100,  0, 1024)
roi_grp.add("roi_dy", int_t,    0, "Region of Interest height",          100,  0, 1024)

green_mask = gen.add_group("GreenMask", type="tab")
green_mask.add("g_hc",   int_t, 0, "Green Mask Hue center",       60,  45,  75)
green_mask.add("g_hs",   int_t, 0, "Green Mask Hue sensitivity",  15,   0,  30)
green_mask.add("g_smin", int_t, 0, "Green Mask Sat min",         100,   0, 255)
green_mask.add("g_smax", int_t, 0, "Green Mask Sat max",         255,   0, 255)
green_mask.add("g_vmin", int_t, 0, "Green Mask Val min",         100,   0, 255)
green_mask.add("g_vmax", int_t, 0, "Green Mask Val max",         255,   0, 255)

red_mask = gen.add_group("RedMask", type="tab")
red_mask.add("r_hc",   int_t, 0, "Red Mask Hue center",      170,  160,200)
red_mask.add("r_hs",   int_t, 0, "Red Mask Hue sensitivity", 15,     0, 30)
red_mask.add("r_smin", int_t, 0, "Red Mask Sat min",        100,     0, 255)
red_mask.add("r_smax", int_t, 0, "Red Mask Sat max",        255,     0, 255)
red_mask.add("r_vmin", int_t, 0, "Red Mask Val min",         85,     0, 255)
red_mask.add("r_vmax", int_t, 0, "Red Mask Val max",        255,     0, 255)


display_mode_enum = gen.enum([ gen.const("None",      int_t, 0, "None"),
                               gen.const("Input",     int_t, 1, "Input"),
                               gen.const("ROI",       int_t, 2, "ROI"),
                               gen.const("GreenMask", int_t, 3, "GreenMask"),
                               gen.const("RedMask",   int_t, 4, "RedMask"),
                               gen.const("Contour",   int_t, 5, "Contour")],
                             "An enum to set display mode")
   
gen.add("display_mode", int_t, 0, "Display mode", 0, 0, 5, edit_method=display_mode_enum)
gen.add("show_hud", bool_t, 0, "Show_HUD", False)
   
exit(gen.generate(PACKAGE, "two_d_guidance", "trr_vision_traffic_light"))
